RecoLocalMuon/GEMRecHit/plugins/GEMRecHitProducer.cc

```diff
@@ -0,0 +1,172 @@
+
+#include "RecoLocalMuon/GEMRecHit/interface/GEMRecHitAlgoFactory.h"
+#include "DataFormats/GEMRecHit/interface/GEMRecHitCollection.h"
+
+#include "CondFormats/DataRecord/interface/GEMMaskedStripsRcd.h"
+#include "CondFormats/DataRecord/interface/GEMDeadStripsRcd.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <string>
+#include <fstream>
+
+using namespace edm;
+using namespace std;
+
+GEMRecHitProducer::GEMRecHitProducer(const ParameterSet& config):
+  theGEMDigiToken(consumes<GEMDigiCollection>(config.getParameter<InputTag>("gemDigiLabel"))),
+  // Get the concrete reconstruction algo from the factory
+  theAlgo{GEMRecHitAlgoFactory::get()->create(config.getParameter<string>("recAlgo"),
+                                              config.getParameter<ParameterSet>("recAlgoConfig"))},
+  maskSource_(MaskSource::EventSetup), deadSource_(MaskSource::EventSetup)
+{
+  produces<GEMRecHitCollection>();
+
+  // Get masked- and dead-strip information from file
+  applyMasking_ = config.getParameter<bool>("applyMasking");
+  if (applyMasking_) {
+    if (config.existsAs<edm::FileInPath>("maskFile", false)) {
+      std::ifstream inputFile(config.getUntrackedParameter<edm::FileInPath>("maskFile").fullPath());
+      if ( !inputFile ) {
+        throw cms::Exception("GEMRecHitProducer")
+          << "Masked Strips File cannot not be opened";
+      }
+      theGEMMaskedStripsObj = std::make_unique<GEMMaskedStrips>();
+      while ( inputFile.good() ) {
+        GEMMaskedStrips::MaskItem Item;
+        inputFile >> Item.rawId >> Item.strip;
+        if ( inputFile.good() ) theGEMMaskedStripsObj->fillMaskVec(Item);
+      }
+      inputFile.close();
+    }
+    
+    if (config.existsAs<edm::FileInPath>("deadFile", false)) {
+      std::ifstream inputFile(config.getUntrackedParameter<edm::FileInPath>("deadFile").fullPath());
+      if ( !inputFile ) {
+        throw cms::Exception("GEMRecHitProducer")
+          << "Dead Strips File cannot not be opened";
+      }
+      theGEMDeadStripsObj = std::make_unique<GEMDeadStrips>();
+      while ( inputFile.good() ) {
+        GEMDeadStrips::DeadItem Item;
+        inputFile >> Item.rawId >> Item.strip;
+        if ( inputFile.good() ) theGEMDeadStripsObj->fillDeadVec(Item);
+      }
+      inputFile.close();
+    }
+  }
+}
+
+GEMRecHitProducer::~GEMRecHitProducer() = default;
+
+void GEMRecHitProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions)
+{
+  edm::ParameterSetDescription desc;
+  edm::ParameterSetDescription recAlgoConfigDesc;
+  desc.add<edm::ParameterSetDescription>("recAlgoConfig", recAlgoConfigDesc);
+  desc.add<std::string>("recAlgo", "GEMRecHitStandardAlgo");
+  desc.add<edm::InputTag>("gemDigiLabel", edm::InputTag("muonGEMDigis"));
+  desc.add<bool>("applyMasking", false);
+  desc.addOptionalUntracked<edm::FileInPath>("maskFile");  
+  desc.addOptionalUntracked<edm::FileInPath>("deadFile");  
+  descriptions.add("gemRecHits",desc);
```

Comment:
In https://github.com/cms-sw/cmssw/pull/25993/files#diff-36be06663e10f5daaf6dda01f26156eaR3 you link a  gemRecHitsDef, then the same name should be defined here
```C++
  descriptions.add("gemRecHitsDef",desc);
```